apps/research-service/src/polygon-clipping/polygon-clipping.controller.ts

```diff
@@ -5,108 +5,189 @@ import {
   })
+  @JsonApiResponse(DelayedJobDto)
   @ExceptionResponse(UnauthorizedException, { description: "Authentication failed." })
   @ExceptionResponse(NotFoundException, { description: "Site not found or no fixable overlapping polygons." })
   @ExceptionResponse(BadRequestException, { description: "Invalid request data." })
-  async createSitePolygonClipping(@Param("siteUuid") siteUuid: string) {
-    await this.policyService.authorize("readAll", SitePolygon);
+  async createSiteClippedVersions(@Param("siteUuid") siteUuid: string, @Request() { authenticatedUserId }) {
+    await this.policyService.authorize("update", SitePolygon);
+
+    const userId = this.policyService.userId;
+    if (userId == null) {
+      throw new UnauthorizedException("User must be authenticated");
+    }
+
+    const user = await User.findByPk(userId, {
+      include: [{ association: "roles", attributes: ["name"] }]
+    });
+    const source = user?.getSourceFromRoles() ?? "terramatch";
+    const userFullName = user?.fullName ?? null;
 
     const fixablePolygons = await this.clippingService.getFixablePolygonsForSite(siteUuid);
 
     if (fixablePolygons.length === 0) {
       throw new NotFoundException(`No fixable overlapping polygons found for site ${siteUuid}`);
     }
 
-    const originalGeometries = await this.clippingService.getOriginalGeometriesGeoJson(fixablePolygons);
+    const site = await Site.findOne({
+      where: { uuid: siteUuid },
+      attributes: ["id", "name"]
+    });
 
-    const clippedResults = await this.clippingService.clipPolygons(fixablePolygons);
-    const clippedGeometries = this.clippingService.buildGeoJsonResponse(clippedResults);
+    if (site == null) {
+      throw new NotFoundException(`Site with UUID ${siteUuid} not found`);
+    }
 
-    return {
-      originalGeometries,
-      clippedGeometries,
-      summary: {
-        totalPolygonsProcessed: fixablePolygons.length,
-        polygonsClipped: clippedResults.length,
-        message: `Successfully processed ${fixablePolygons.length} polygons, clipped ${clippedResults.length} polygons`
+    const delayedJob = await DelayedJob.create({
+      isAcknowledged: false,
+      name: "Polygon Clipping",
+      totalContent: fixablePolygons.length,
+      processedContent: 0,
+      progressMessage: "Starting clipping...",
+      createdBy: authenticatedUserId,
+      metadata: {
+        entity_id: site.id,
+        entity_type: "App\\Models\\V2\\Sites\\Site",
+        entity_name: site.name
       }
-    };
+    } as DelayedJob);
+
+    await this.clippingQueue.add("clipAndVersion", {
+      polygonUuids: fixablePolygons,
+      userId,
+      userFullName,
+      source,
+      delayedJobId: delayedJob.id
+    });
+
+    return buildJsonApi(DelayedJobDto).addData(delayedJob.uuid, new DelayedJobDto(delayedJob));
   }
 
-  @Post("projects/:siteUuid/clippedPolygons")
+  @Post("projects/:siteUuid/clippedVersions")
```

Comment:
It's definitely confusing to have a `sites/:siteUuid` endpoint and a `projects/:siteUuid` endpoint.
* Why are both needed?
* Why is this path under projects, but takes a site UUID? That definitely breaks convension.

If a projects endpoint is really needed, please have it accept the project UUID instead. There's also a ton of duplication between these methods. Consider creating a private utility method to handle the duplication in this controller, or maybe a different path structure would be better, with a single endpoint. Something like 
`clippedVersions` with `siteUuid` and `projectUuid` query params.